library/Reporting/Model/Report.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/* Icinga Reporting | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Reporting\Model;
+
+use ipl\Orm\Behavior\MillisecondTimestamp;
+use ipl\Orm\Behaviors;
+use ipl\Orm\Model;
+use ipl\Orm\Relations;
+
+class Report extends Model
+{
+    public function getTableName()
+    {
+        return 'report';
+    }
+
+    public function getKeyName()
+    {
+        return 'id';
+    }
+
+    public function getColumns()
+    {
+        return [
+            'timeframe_id',
+            'template_id',
+            'author',
+            'name',
+            'ctime',
+            'mtime'
+        ];
+    }
+
+    public function getDefaultSort()
+    {
+        return ['name'];
+    }
+
+    public function createBehaviors(Behaviors $behaviors)
+    {
+        $behaviors->add(new MillisecondTimestamp([
+            'ctime',
+            'mtime'
+        ]));
+    }
+
+    public function createRelations(Relations $relations)
+    {
+        $relations->belongsTo('timeframe', Timeframe::class);
+        $relations->belongsTo('template', Template::class)
+            ->setJoinType('LEFT');
```

Comment:
Why `LEFT JOIN`?

Reply:
It is optional, i.e you can have reports without template, so it can't be used `INNER JOIN`.